app/_development-tools/controllers.js

```diff
@@ -0,0 +1,27 @@
+const { uniq } = require('lodash')
+
+const { permissionsByRole } = require('../../common/lib/permissions')
+
+function renderPermissions(req, res) {
+  res.render('_development-tools/views/permissions', {
+    activeRoles: req.session.activeRoles || [],
+    roles: permissionsByRole,
+  })
+}
+
+function updatePermissions(req, res) {
```

Comment:
Future enhancement idea - supporting updating by both form and JSON.

We used this on session injection to give humans an easy form, but then used the JSON updating as a more reliable approach for automated browser tests

Reply:
Yep, that's a really nice idea! Nothing preventing us extending this to support a JSON body, or in fact just a `POST` within the e2e tests?